modules/Bio/EnsEMBL/Variation/Utils/VariationEffect.pm

```diff
@@ -347,6 +347,7 @@ sub _after_end {
 
 sub _upstream {
     my ($bvf, $feat, $dist) = @_;
+    $dist += get_max_shift_length($bvf);
```

Comment:
is this something that can be cached?

Reply:
Unfortunately not, if a new shift hash is created then this value could change, the defensive thing to do is to calculate it every time